internal/pkg/cli/pipeline_init.go

```diff
@@ -137,43 +138,52 @@ func newInitPipelineOpts(vars initPipelineVars) (*initPipelineOpts, error) {
 		sel:              selector.NewSelect(prompter, ssmStore),
 		runner:           exec.NewCmd(),
 		fs:               &afero.Afero{Fs: afero.NewOsFs()},
+		wsAppName:        tryReadingAppName(),
 	}, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error if the optional flag values passed by the user are invalid.
 func (o *initPipelineOpts) Validate() error {
-	if o.appName == "" {
+	// This command must be executed in the app's workspace because the pipeline manifest and buildspec will be created and stored.
+	if o.wsAppName == "" {
 		return errNoAppInWorkspace
 	}
-	if _, err := o.store.GetApplication(o.appName); err != nil {
-		return err
+	if o.appName != "" {
+		if o.appName != o.wsAppName {
+			return fmt.Errorf("cannot specify app %s because the workspace is already registered with app %s", o.appName, o.wsAppName)
+		}
+		// Validate the app name.
+		if _, err := o.store.GetApplication(o.appName); err != nil {
+			return fmt.Errorf("get application %s configuration: %w", o.appName, err)
+		}
 	}
+	o.appName = o.wsAppName
```

Comment:
nit: it's surprising that an assignment happens in `Validate` can we do it in the constructor instead?
```go
wsAppName := tryReadingAppName()
if vars.appName == "" {
   vars.appName = wsAppName
}
```

Reply:
Hmm but that means we would have to move the validation above there as well because we need the original value they pass in via the flag for validation. If we override like this then there's no way for us to tell them there's something wrong with their app flag value

Reply:
But we wouldn't override if they pass in a value....